src/utils/create-entity-api-utils.ts

```diff
@@ -156,8 +204,23 @@ export const createEntityApiUtils = <
         dispatch(api.util.upsertQueryData('get', { id: entity.id, params: entityRequest }, entity));
       }
     },
+    /**
+     * Handles entity update.
+     * Uses patchEntityQueries under hood and intended to be used in `onQueryStarted` callback to perform optimistic/pessimistic update of entity data in queries connected by tags.
+     * If `optimistic` flag is set to true, it performs optimistic patching.
+     *
+     * @param {EntityPartial<TEntity> | TEntity['id']} arg - The entity data or id.
+     * @param {Object} endpointLifecycle - The endpoint lifecycle.
+     * @param {Object} endpointLifecycle.dispatch - The dispatch function.
+     * @param {Object} endpointLifecycle.getState - The getState function.
+     * @param {Object} [options] - The options.
+     * @param {boolean} [options.optimistic=false] - Whether to perform optimistic patching.
+     * @param {boolean} [options.shouldRefetchEntity=false] - Whether to refetch the entity.
+     * @param {Array<TagDescription>} [options.tags] - The tags.
+     * @returns {Promise<void>} The promise.
```

Comment:
I think if a function returns `void` or  `Promise<void>` there is nothing to comment

```suggestion
     * @returns {Promise<void>}
```